File: java/Animal.java
public class Animal {

    private int id; 
    public  String nombre; 
    protected  int edad; 

    public Animal (int id, String nombre, int edad) {
        this.id = id; 
        this.nombre = nombre; 
        this.edad  = edad; 

    }
public int getId () {
    return this.id; 

}
public String getnombre() {
    return this.nombre; 
}
public int getEdad () {
    return this.edad; 

}
public void setId (int nuevoid) {
    this.id = nuevoid; 

}
public void setNombre (String nuevonombre) {
    this.nombre = nuevonombre; 
}
public void setEdad (int nuevoEdad ) {
    this.edad = nuevoEdad; 
}

public void mostrarDatos  () {
    System.out.println("id: " + id + "nombre: " + nombre + "edad:" + edad);
}
public void  comer (String come) {
  System.out.println("Animal: " + nombre + " le gusta comer " + come);
  
}
public int sumar (int num1, int num2) {
    int suma = num1 + num2; 
    return suma; 
}

public String toJSON() {
    return "{" + "id : " + id + " Nombre : " + nombre + "edad: " + edad + "}"; 

    
}


    
}

File: Go/Animal.go

type Animal struct {
	id     int
	nombre string
	edad   int
}


func NewAnimal(id int, nombre string, edad int) Animal {
	return Animal{id: id, nombre: nombre, edad: edad}
}


func (a Animal) GetId() int {
	return a.id
}

func (a Animal) GetNombre() string {
	return a.nombre
}

func (a Animal) GetEdad() int {
	return a.edad
}


func (a *Animal) SetId(nuevoId int) {
	a.id = nuevoId
}

func (a *Animal) SetNombre(nuevoNombre string) {
	a.nombre = nuevoNombre
}

func (a *Animal) SetEdad(nuevaEdad int) {
	a.edad = nuevaEdad
}


func (a Animal) MostrarDatos() {
	fmt.Println("id:", a.id, "nombre:", a.nombre, "edad:", a.edad)
}


func (a Animal) Comer(comida string) {
	fmt.Println("Animal:", a.nombre, "le gusta comer", comida)
}


func (a Animal) Sumar(num1 int, num2 int) int {
	return num1 + num2
}


func (a Animal) ToJSON() string {
	return fmt.Sprintf("{id: %d, nombre: %s, edad: %d}", a.id, a.nombre, a.edad)
}